system/lib/pthread/library_pthread.c

```diff
@@ -584,7 +592,7 @@ void emscripten_current_thread_process_queued_calls() {
     // Assume that the call is heavy, so unlock access to the call queue while it is being
     // performed.
     pthread_mutex_unlock(&call_queue_lock);
-    _do_call(q->call_queue[head]);
+    q->call_queue[head].func(q->call_queue[head].arg);
```

Comment:
It worries me a little to do two subsequent reads here without holding the call queue lock. I think it should be safe as a single one was apparently safe to do before, and I think that is the case because this is the only place that removes things from the call queue - other threads only add, and they add after the head. Is that right?

Reply:
Yes, I think that's right. Notably the call queue head has not been updated yet, so any other thread trying to add things will see this as an occupied slot and will not write to it.